### DISCUSSION TITLE
Help requested about dhcpcd running mode when configuring statically the system interfaces

### DISCUSSION THREAD

**Discussion by quilix** [rsmarples/dhcpcd#234] [Q&A] [answered]
Dhcpcd can configure network interfaces with information provided by
a DHCP server or with static information provided in the configuration
file dhcpcd.conf.

In my actual situation I want to use a single statically configured
interface and I use the following configuration:

```
interface eth0
arping 192.168.1.1

profile 20:b0:01:36:78:fc
    persistent
    static ip_address=192.168.1.2/24
    static broadcast_address=192.168.1.255
    static routers=192.168.1.1
    static domain_name_servers=192.168.1.1
```
Now, at boot time, dhcpcd is started with the following command line:

dhcpcd -b -q

What happens is the following:

1. dhcpcd continues to run in background.
2. dhcpcd opens the TPC port 68.

I have the following questions:

1) when all interfaces are statically configured what is the purpose
 of having dhcpcd running in background? It doesn't have to monitor
 expiring leases and to renew them. Are there other reasons for which
 dhcpcd should be left running in background?

2) when all interfaces are statically configured what is the purpose of opening the port 68?
   
3) is there a mode to start dhcpcd and tell it to exit when configuring
 only statically interfaces? There is the -1 option which does something
 like this but I am not sure about its purpose (for example this option
 cannot be used to configure two statically configured interfaces)

Thank you
Cristian

**Comment by ido** [comment, marked as answer]
If you don't need a DHCP client because you're statically configuring all of your interfaces, then running `dhcpcd` is probably not necessary in the first place, and you should investigate what other network configuration tools your OS/distro provides. You're just using `dhcpcd` as a wrapper to `ip addr` / `ifconfig` on startup here if I understood you correctly. (If you're on Linux and your distro uses `systemd`, you can configure interfaces statically through `systemd` as well, for example.). Hope this helps, reopen if you have further questions.
